kernel/build-kernel.sh

```diff
@@ -0,0 +1,252 @@
+
+check_version_yaml()
+{
+	info "check versions.yaml"
+	local versions_yaml="${GOPATH}/src/${runtime_repository}/versions.yaml"
+	if [ ! -f "$versions_yaml" ]; then
+		info "Download ${runtime_repository}"
+		go get "${runtime_repository}"
+	fi
+	[ -f "$versions_yaml" ] || die "failed to found ${versions_yaml}"
+	
+
+}
+
+get_kernel()
+{
+	local version="${1}"
+	#Remove extra 'v'
+	version=${version#v}
+
+	info "Download kernel version ${version}"
+
+	major_version=$(echo "${version}" | cut -d. -f1)
+	kernel_tarball="linux-${version}.tar.xz"
+
+	if [ ! -f "${kernel_tarball}" ];then
+		info "Download kernel"
+		curl -OL "https://www.kernel.org/pub/linux/kernel/v${major_version}.x/${kernel_tarball}"
+	else
+		info "kernel tarball already downloaded"
+	fi
+
+	curl -OL "https://cdn.kernel.org/pub/linux/kernel/v${major_version}.x/sha256sums.asc"
+	grep "${kernel_tarball}" sha256sums.asc > "${kernel_tarball}.sha256"
+
+	sha256sum -c "${kernel_tarball}.sha256"
+
+	tar xf ${kernel_tarball}
+
+	# update global variable 
+	kernel_path=$(cd "linux-${version}" && pwd)
+
+
+}
+
+get_default_kernel_config(){
+	local version="${1}"
+	#Remove extra 'v'
+	local version="${version#v}"
+
+	local hypervisor="$2"
+	local kernel_arch="$3"
+
+	[ -n "${version}" ] || die "kernel version not provided"
+	[ -n "${hypervisor}" ] || die "hypervisor not provided"
+	[ -n "${kernel_arch}" ] || die "kernel arch not provided"
+
+	major_version=$(echo "${version}" | cut -d. -f1)
+	minor_version=$(echo "${version}" | cut -d. -f2)
+	config="${default_kernel_config_dir}/${kernel_arch}_kata_${hypervisor}_${major_version}.${minor_version}.x"
+	[ -f "${config}" ] || die "failed to find default config ${config}"
+	echo "${config}"
+}
+
+get_config_version(){
+	config_version_file="${default_patches_dir}/../kata_config_version"
+	if [ -f "${config_version_file}" ];then
+		cat "${config_version_file}"
+	else
+		echo "unknown"
+	fi
+}
+
+main()
+{
+	while getopts "a:c:hk:t:p:v:" opt
+	do
+	case "$opt" in
+			a)
+				arch_target="${OPTARG}"
+				;;
+			c)
+				kernel_config_path="${OPTARG}"
+				;;
+
+			h)
+				usage
+				exit 0
+				;;
+
+			k)
+				kernel_path="${OPTARG}"
+				exit 0
+				;;
+
+			t)
+				hypervisor_target="${OPTARG}"
+				;;
+			p)
+				patches_path="${OPTARG}"
+				;;
+			v)
+				kernel_version="${OPTARG}"
+				;;
+			esac
+	done
+
+	shift $[$OPTIND-1]
+
+
+	if [ -z "$kernel_path" ]; then
+		info "kernel path not provided, will download kernel"
+		download_kernel="true"
+		# If not kernel version take it from versions.yaml
+		if [ -z "$kernel_version" ]; then
+			check_version_yaml && kernel_version=$(get_from_kata_deps "assets.kernel.version")
+		fi
+	fi
+	[ -n "$kernel_version" ] ||  die "failed to get kernel version: Kernel version is emtpy"
+
+	if [[ "${download_kernel}" == "true" ]];then
+		get_kernel "${kernel_version}"
+	fi
+
+	[ -n "$kernel_path" ] ||  die "failed to find kernel source path"
+
+	[ -n "${patches_path}" ] || patches_path="${default_patches_dir}"
```

Comment:
If patches_path is not provided, then it would `go get` the packaging repo, right? What if you are running from the repo itself.
Can we download the patches insead to a directory created by this script? The reason is I would like to see no dependency on golang in this script, one should be able to run this script standalone without the assumption of a go environment setup.